src/raftstore/store/engine.rs

```diff
@@ -240,11 +240,29 @@ pub trait Mutable: Writable {
 impl Mutable for DB {}
 impl Mutable for WriteBatch {}
 
+// `delete_in_range` fast deletes date in range [start_key, end_key).
+// It uses rocksdb `delete_file_in_range` first, then scans the left keys and
+// uses WriteBatch to deletes them.
+// Note: this function is dangerous and not guarantees consistence. If you call
+// delete_file_in_range successfully but commit following WriteBatch failed,
+// some keys are really deleted and can't be recovered.
+pub fn delete_in_range(db: &DB, wb: &WriteBatch, start_key: &[u8], end_key: &[u8]) -> Result<()> {
```

Comment:
yes, delete_file_in_range can't guarantee deleting all data in [start, end), we must scan to delete left keys.